### PR TITLE
[connectors] - feat: distinct webcrawling queues

### LINKED ISSUE #5015: Add jitter to website refresh workflows
Updates to websites come as spikes which starve the aggregation of new websites

Easy solution is to smooth websites update on 8 attempt (1 per hour)
More complex solution is to create a new queue for new websites

We may want to also increase concurrency and maybe decrease the upsert queue concurrency

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

Treat newly added websites and websites to update in different task queues

## Risk

Not much

### CODE REVIEW

File: connectors/src/connectors/webcrawler/temporal/client.ts [hunk 1]
```diff
@@ -16,15 +16,15 @@ import {
 } from "./workflows";
 
 export async function launchCrawlWebsiteWorkflow(
-  connectorId: ModelId
+  connectorId: ModelId,
+  priorityLevel: number | null
```
Comment (lasryaric) [inline connectors/src/connectors/webcrawler/temporal/client.ts:20]: priorityLevel could be replaced by priority and a typescript type of `"new_website" | "refresh_website" to be very explicit. Update the call sites accordingly.

**Review by lasryaric — COMMENTED** [review]
Much better. Left two comments 👍

File: connectors/src/connectors/webcrawler/temporal/client.ts (hunk 1 above)
Comment (JulesBelveze) [reply connectors/src/connectors/webcrawler/temporal/client.ts:20]: @lasryaric I used `number | null` to meet the `SyncConnector` signature

File: connectors/src/connectors/webcrawler/index.ts [hunk 2]
```diff
@@ -443,10 +443,27 @@ export async function setWebcrawlerConfiguration(
   if (stopRes.isErr()) {
     return new Err(stopRes.error);
   }
-  const startRes = await launchCrawlWebsiteWorkflow(connector.id);
+
+  const queueName = webcrawlerConfig.lastCrawledAt ? 0 : 1;
```
Comment (spolu) [inline connectors/src/connectors/webcrawler/index.ts:447]: queueName is definitely ill named here. Agree with @lasryaric that we want a clean type at the client interface

**Review by spolu — COMMENTED** [review]
Splitting queues allows not starving the new websites which is great.
ooc what is our plan to smooth a bit the refreshes?

**Comment by JulesBelveze** [comment]
@spolu From what I gathered after discussions with the team the main issue here was new websites starvation, while refresh website was not a big deal. This is why I introduced this new queue.

However if you think we should still smooth out the refreshes I can think about it and come up with a plan?

**Comment by lasryaric** [comment]
>ooc what is our plan to smooth a bit the refreshes?

Since we have two workers now (first crawling and refresh), we can set the activity concurrency of each worker separately, which allows us to smooth out refreshs.

**Comment by spolu** [comment]
> > ooc what is our plan to smooth a bit the refreshes?
> 
> Since we have two workers now (first crawling and refresh), we can set the activity concurrency of each worker separately, which allows us to smooth out refreshs.

Well we'll still have spikes right?

I agree it might not need to be addressed right now as we can smooth it at the upsert queue level if we desire so.

**Review by JulesBelveze — COMMENTED** [review]
@spolu Took the freedom to let the client assign the workflow to the task queue. Let me know what u think

**Review by spolu — APPROVED** [review]
LGTM

modulo dead code :)

File: connectors/src/connectors/webcrawler/index.ts [hunk 3]
```diff
@@ -443,10 +443,26 @@ export async function setWebcrawlerConfiguration(
...
+
   const startRes = await launchCrawlWebsiteWorkflow(connector.id);
   if (startRes.isErr()) {
     return new Err(startRes.error);
   }
 
   return new Ok(undefined);
 }
+
+export async function isNewWebsiteToCrawl(
```
[the remark this answers is not in the document]
Comment (JulesBelveze) [reply connectors/src/connectors/webcrawler/index.ts]: Not anymore 👍🏼

**Comment by lasryaric** [comment]
> Well we'll still have spikes right?

@spolu we'll still have spike of enqueued workflow in the "refresh" queue, but not spikes of processing workflows, which I think it the right design.

**Pull request merged**